server/src/domains/users/users.controller.ts

```diff
@@ -0,0 +1,67 @@
+import { Controller, Get, Post, Body, Param, Delete, Put, UseGuards, UsePipes, ValidationPipe } from '@nestjs/common';
+import { UsersService } from './users.service';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { AuthGuard } from '../auth/auth.guard';
+import { UserPayload } from '../auth/dto/userPayload.dto';
+import { UserRequest } from './userRequest.decorator';
+import { AuthService } from '../auth/auth.service';
+import { User } from './entities/user.entity';
+
+@Controller('users')
+export class UsersController {
+  constructor(private readonly usersService: UsersService,
+    private authService: AuthService) { }
+
+  @UseGuards(AuthGuard)
+  @Post()
+  async create(@Body() createUserDto: CreateUserDto): Promise<User> {
+    return await this.usersService.create(createUserDto);
+  }
+
+  @UseGuards(AuthGuard)
+  @Get('all')
+  async findAll(): Promise<User[]> {
+    return await this.usersService.getAll();
+  }
+
+  @UseGuards(AuthGuard)
+  @Get(':id')
+  async findOneByAdmin(@Param('id') id: number): Promise<User> {
```

Comment:
Do not use business names like 'Admin' here. If you want to mark that this route will be protected  it is better to add a comment, but it should be self-explanatory with using guards.